[generic type] Regression in being able to spoon Guava

It seems that #1218 introduced a regression on Guava project which is used by our CI to check the behaviour of Spoon, see the trace : 
<details>

> Cloning the remote Git repository
> Cloning repository https://github.com/spoon-bot/guava.git
> git init /builds/workspace/Guava # timeout=10
> Fetching upstream changes from https://github.com/spoon-bot/guava.git
> git --version # timeout=10
> git fetch --tags --progress https://github.com/spoon-bot/guava.git +refs/heads/*:refs/remotes/origin/*
> git config remote.origin.url https://github.com/spoon-bot/guava.git # timeout=10
> git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
> git config remote.origin.url https://github.com/spoon-bot/guava.git # timeout=10
> Fetching upstream changes from https://github.com/spoon-bot/guava.git
> git fetch --tags --progress https://github.com/spoon-bot/guava.git +refs/heads/*:refs/remotes/origin/*
> git rev-parse refs/remotes/origin/master^{commit} # timeout=10
> git rev-parse refs/remotes/origin/origin/master^{commit} # timeout=10
> Checking out Revision cc335985ccd19c6dd13087ca3c13c92e7793c36b (refs/remotes/origin/master)
> git config core.sparsecheckout # timeout=10
> git checkout -f cc335985ccd19c6dd13087ca3c13c92e7793c36b
> git rev-list cc335985ccd19c6dd13087ca3c13c92e7793c36b # timeout=10
> [Guava] $ /bin/sh -xe /tmp/hudson8643585831120465795.sh
> rm -rf /builds/.m2/repository/fr/inria/gforge/spoon
> [Guava] $ /bin/sh -xe /tmp/hudson1465629534996861412.sh
> [...]
> [INFO] ------------------------------------------------------------------------
> [INFO] BUILD FAILURE
> [INFO] ------------------------------------------------------------------------
> [INFO] Total time: 1:21.906s
> [INFO] Finished at: Tue Apr 25 01:32:31 CEST 2017
> [INFO] Final Memory: 22M/644M
> [INFO] ------------------------------------------------------------------------
> [ERROR] Failed to execute goal fr.inria.gforge.spoon:spoon-maven-plugin:2.4:generate (default) on project guava: Exception during the spoonify of the target project. spoon.SpoonException: Cannot resolve 1) type parameter <K>  of declarer @com.google.common.annotations.GwtIncompatible
> [ERROR] public abstract class ForwardingNavigableMap<K, V> extends com.google.common.collect.ForwardingSortedMap<K, V> implements java.util.NavigableMap<K, V> {
> [ERROR] protected ForwardingNavigableMap() {
> [ERROR] }
> [ERROR] 
> [ERROR] @java.lang.Override
> [ERROR] protected abstract java.util.NavigableMap<K, V> delegate();
> [ERROR] 
> [ERROR] @java.lang.Override
> [ERROR] public java.util.Map.Entry<K, V> lowerEntry(K key) {
> [ERROR] return delegate().lowerEntry(key);
> [ERROR] }
> [ERROR] 
> [ERROR] protected java.util.Map.Entry<K, V> standardLowerEntry(K key) {
> [ERROR] return headMap(key, false).lastEntry();
> [ERROR] }
> [ERROR] 
> [ERROR] @java.lang.Override
> [ERROR] public K lowerKey(K key) {
> [ERROR] return delegate().lowerKey(key);
> [ERROR] }
> [...]

</details>

See full stack trace:
https://ci.inria.fr/sos/job/Guava/262/console

--- a/doc/jenkins/build.sh
+++ b/doc/jenkins/build.sh
@@ -172,7 +172,7 @@ xmlstarlet ed -N x="http://maven.apache.org/POM/4.0.0" -s "/x:project/x:build/x:
 xmlstarlet ed -N x="http://maven.apache.org/POM/4.0.0" -s "/x:project/x:build/x:plugins/x:plugin[last()]/x:dependencies" --type elem -n dependency -v "" pom.bak14.xml > pom.bak19.xml
 xmlstarlet ed -N x="http://maven.apache.org/POM/4.0.0" -s "/x:project/x:build/x:plugins/x:plugin[last()]/x:dependencies/x:dependency[last()]" --type elem -n groupId -v "fr.inria.gforge.spoon" pom.bak19.xml > pom.bak20.xml
 xmlstarlet ed -N x="http://maven.apache.org/POM/4.0.0" -s "/x:project/x:build/x:plugins/x:plugin[last()]/x:dependencies/x:dependency[last()]" --type elem -n artifactId -v "spoon-core" pom.bak20.xml > pom.bak21.xml
-xmlstarlet ed -N x="http://maven.apache.org/POM/4.0.0" -s "/x:project/x:build/x:plugins/x:plugin[last()]/x:dependencies/x:dependency[last()]" --type elem -n version -v "[5.5.0-SNAPSHOT,)" pom.bak21.xml > pom.bak22.xml
+xmlstarlet ed -N x="http://maven.apache.org/POM/4.0.0" -s "/x:project/x:build/x:plugins/x:plugin[last()]/x:dependencies/x:dependency[last()]" --type elem -n version -v "[7.0.0-SNAPSHOT,)" pom.bak21.xml > pom.bak22.xml
 mv pom.bak22.xml pom.xml
 rm pom.bak*.xml
 

--- a/doc/jenkins/build.sh
+++ b/doc/jenkins/build.sh
@@ -155,7 +155,11 @@ rm pom.bak*.xml
 xmlstarlet ed -N x="http://maven.apache.org/POM/4.0.0" -s "/x:project/x:build/x:plugins" --type elem -n plugin -v "" pom.xml > pom.bak.xml
 xmlstarlet ed -N x="http://maven.apache.org/POM/4.0.0" -s "/x:project/x:build/x:plugins/x:plugin[last()]" --type elem -n groupId -v "fr.inria.gforge.spoon" pom.bak.xml > pom.bak2.xml
 xmlstarlet ed -N x="http://maven.apache.org/POM/4.0.0" -s "/x:project/x:build/x:plugins/x:plugin[last()]" --type elem -n artifactId -v "spoon-maven-plugin" pom.bak2.xml > pom.bak3.xml
-xmlstarlet ed -N x="http://maven.apache.org/POM/4.0.0" -s "/x:project/x:build/x:plugins/x:plugin[last()]" --type elem -n version -v "2.4" pom.bak3.xml > pom.bak4.xml
+
+# we depend on the latest version of spoon-maven-plugin, one that does not use http://spoon.gforge.inria.fr/repositories/snapshots/ (decommissioned)
+# but the correct http://maven.inria.fr/artifactory/spoon-public-snapshot/ (Inria's artifactory) 
+xmlstarlet ed -N x="http://maven.apache.org/POM/4.0.0" -s "/x:project/x:build/x:plugins/x:plugin[last()]" --type elem -n version -v "3.1" pom.bak3.xml > pom.bak4.xml
+
 xmlstarlet ed -N x="http://maven.apache.org/POM/4.0.0" -s "/x:project/x:build/x:plugins/x:plugin[last()]" --type elem -n executions -v "" pom.bak4.xml > pom.bak5.xml
 xmlstarlet ed -N x="http://maven.apache.org/POM/4.0.0" -s "/x:project/x:build/x:plugins/x:plugin[last()]/x:executions" --type elem -n execution -v "" pom.bak5.xml > pom.bak6.xml
 xmlstarlet ed -N x="http://maven.apache.org/POM/4.0.0" -s "/x:project/x:build/x:plugins/x:plugin[last()]/x:executions/x:execution" --type elem -n phase -v "generate-sources" pom.bak6.xml > pom.bak7.xml

